Sources/StreamChat/WebSocketClient/WebSocketClient.swift

```diff
@@ -65,12 +65,13 @@ class WebSocketClient {
         return pingController
     }()
 
-    private func createEngineIfNeeded(for connectEndpoint: Endpoint<EmptyResponse>) -> WebSocketEngine {
+    private func createEngineIfNeeded(for connectEndpoint: Endpoint<EmptyResponse>) throws -> WebSocketEngine {
         let request: URLRequest
         do {
             request = try requestEncoder.encodeRequest(for: connectEndpoint)
         } catch {
-            fatalError("Failed to create WebSocketEngine with error: \(error)")
+            log.log(.debug, message: error.localizedDescription)
```

Comment:
Why not `log.error(error.localizedDescription, subsystems: .websocket)`?

Reply:
I didn't want to pollute the console with an error that is recoverable. In rare cases, this will happen, but it won't have any impact. So it should not warn the customer that something is wrong.

Reply:
But I'm also not sure on this one 🤔

Reply:
Just to be safe, and since this will only happen 99% of times, I will change to `.error` 👍

Reply:
If we are not handling the error higher, we should at least log something so customers have something to pinpoint to a potential issue